src/passes/pass.cpp

```diff
@@ -192,7 +193,7 @@ void PassRunner::run() {
       std::cerr << "final module does not validate\n";
       abort();
     }
-    passDebug = currPassDebug;
+    recursion--;
```

Comment:
Oh, I didn't read this closely enough before to realize we *were* writing to the static var. That seems less-than-good. Tracking the recursion instead seems more direct in terms of what it's expressing, but seems similarly fragile.
In particular, we have to remember to never leave this scope between `recursion++` and `recursion--`.

My understanding is that we recurse indirectly, when passes call other passes? If so we can have two functions here, `run` to run passes externally, and `runInternal` that other passes can call.
The high-level idea that's being expressed here seems to be "Am I a top-level pass?".
Quick grepping makes it look like we currently call nested PassRunners in the following way:
```c++
PassRunner passRunner(module);
passRunner.add<FixImports>(&illegalToLegal);
passRunner.run();
```
, where this takes place inside of `void run(PassRunner* runner, Module* module) override`. Would it be possible or desirable to have nested passes reuse the original PassRunner? Something like
```c++
runner->runNestedPass<FixImports>(&illegalToLegal);
```

The key here I think being making explicit the difference between recursive calls and top-level calls at the call site, which lets us avoid a bit of implicit state. Probably the simplest way to do that would be to replace `run()` with `runNested()` for passes called from passes.